main/src/library/List.flix

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2016 Liam Palmer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace List {
+
+    /**
+     * Returns the number of elements in `xs`.
+     */
+    def size[a](xs: List[a]): Int32 = match xs with {
+        case Nil => 0
+        case y :: ys => size(ys) + 1
+    }
+
+    /**
+     * Returns the first element of `xs`.
+     * Throws an exception if `xs` is empty.
+     */
+    def head[a](xs: List[a]): a = match xs with {
+        case Nil => ???
+        case y :: ys => y
+    }
+
+    /**
+     * Returns `xs` with the first element removed.
+     * Throws an exception if `xs` is empty.
+     */
+    def tail[a](xs: List[a]): List[a] = match xs with {
+        case Nil => ???
+        case y :: ys => ys
+    }
+
+    /**
+     * Returns `xs` with `x` prepended to the beginning.
+     */
+    def prepend[a](xs: List[a], x: a): List[a] = x :: xs
+
+    /**
+     * Returns `xs` with `x` appended to the end.
+     */
+    def append[a](xs: List[a], x: a): List[a] = match xs with {
```

Comment:
Append should have the type `def append[a](xs: List[a], ys: List[a]): List[a]`. See the List template file.